File: 04-http/examples/http-client.go
package main

import (
	"fmt"
	"io"
	"net/http"
	"os"
)

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: go run main.go <url>")
		return
	}

	url := os.Args[1]

	resp, err := http.Get(url)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	fmt.Println("Status:", resp.Status)
	fmt.Println("Headers:")
	for key, values := range resp.Header {
		for _, value := range values {
			fmt.Printf("  %s: %s\n", key, value)
		}
	}

	fmt.Println("\nBody:")
	body, _ := io.ReadAll(resp.Body)
	fmt.Println(string(body))
}

File: 04-http/examples/http-client.py
import requests
import sys

def main():
    if len(sys.argv) < 2:
        print("Usage: python main.py <url>")
        return

    url = sys.argv[1]
    response = requests.get(url)

    print("Status:", response.status_code)
    print("Headers:")
    for key, value in response.headers.items():
        print(f"  {key}: {value}")

    print("\nBody:")
    print(response.text)

if __name__ == "__main__":
    main()
